### PR TITLE
Fix exported types and update tooling

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

Types exported by this app would resolve to `any` when other apps tried to use them.

#### Screenshots or example usage:

These are the generated types for`useDevice` (print taken during development of vtex.minicart):

Screen Shot 2020-12-11 at 13 24 28

### CODE REVIEW

**Review by klzns — APPROVED** [review]

File: react/useDevice.tsx [hunk 1]
```diff
@@ -1,16 +1,13 @@
 import { useRuntime } from 'vtex.render-runtime'
+import type { RenderContext } from 'vtex.render-runtime'
+
+export type Device = RenderContext.RenderContext['deviceInfo']['type']
```
Comment (kaisermann) [inline react/useDevice.tsx:4]: Any reason to be exporting this?

File: react/DeviceDetector.tsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
-import React, { FunctionComponent } from 'react'
-import useDevice, { DeviceInfo } from './useDevice'
+import type { FC } from 'react'
+import type { RenderContext } from 'vtex.render-runtime'
+
+import useDevice from './useDevice'
+
+export type Device = RenderContext.RenderContext['deviceInfo']['type']
```
Comment (kaisermann) [inline react/DeviceDetector.tsx:6]: Any reason to be exporting this?

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/useDevice.tsx (hunk 1 above)
Comment (victorhmp) [reply react/useDevice.tsx:4]: Nope. That's unnecessary :)

File: react/DeviceDetector.tsx (hunk 2 above)
Comment (victorhmp) [reply react/DeviceDetector.tsx:6]: Also unnecessary

**Review by kaisermann — APPROVED** [review]

**Review by icazevedo — APPROVED** [review]

**Pull request merged**